src/main/java/io/cdap/plugin/dlp/DLPTransformPluginConfig.java

```diff
@@ -127,113 +127,122 @@ public void validate(FailureCollector collector, Schema inputSchema) {
+        collector.addFailure(String.format("This transform depends on custom template that was not defined.",
+                                           config.getTransform(), String.join(", ", config.getFields())),
+                             "Enable the custom template option and provide the name of it.")
+          .withConfigElement(FIELDS_TO_TRANSFORM, GSON.toJson(errorConfig));
+      }
+      //Validate the config for the transform
+      config.validate(collector, inputSchema, FIELDS_TO_TRANSFORM);
+
+      //Check that custom template and built-in types are not mixed
+      anyTransformUsedCustomTemplate = anyTransformUsedCustomTemplate || filters.contains("NONE");
+      if (firstTransformUsedCustomTemplate == null) {
+        firstTransformUsedCustomTemplate = filters.contains("NONE");
+      } else {
+        if (filters.contains("NONE") != firstTransformUsedCustomTemplate) {
+          errorConfig.setTransformPropertyId("filters");
+          collector.addFailure("Cannot use custom templates and built-in filters in the same plugin instance.",
+                               "All transforms must use custom templates or built-in filters, not a "
+                                 + "combination of both.")
+            .withConfigElement(FIELDS_TO_TRANSFORM, GSON.toJson(errorConfig));
+        }
+      }
 
-          //Checking if location of wrapped key is same as DLP resource location
-          DlpTransformConfig transformProperties = config.getTransformProperties();
-          if (transformProperties instanceof CryptoDeterministicTransformationConfig) {
-            CryptoDeterministicTransformationConfig cryptoDeterministicTransformationConfig =
-              (CryptoDeterministicTransformationConfig) transformProperties;
-            if (cryptoDeterministicTransformationConfig.getKeyType() == KeyType.KMS_WRAPPED) {
-              String cryptoKeyName = cryptoDeterministicTransformationConfig.getCryptoKeyName();
-              CryptoKeyHelper.
-                validateCryptoKeyNameLocation(location, cryptoKeyName, collector, FIELDS_TO_TRANSFORM, LOCATION);
-            }
-          } else if (transformProperties instanceof CryptoHashTransformationConfig) {
-            CryptoHashTransformationConfig cryptoHashTransformationConfig =
-              (CryptoHashTransformationConfig) transformProperties;
-            if (cryptoHashTransformationConfig.getKeyType() == KeyType.KMS_WRAPPED) {
-              String cryptoKeyName = cryptoHashTransformationConfig.getCryptoKeyName();
-              CryptoKeyHelper.
-                validateCryptoKeyNameLocation(location, cryptoKeyName, collector, FIELDS_TO_TRANSFORM, LOCATION);
+      // Make sure the combination of field, transform and filter are unique
+      for (String field : config.getFields()) {
+        for (String filter : config.getFilterDisplayNames()) {
+          String transformKey = String.format("%s:%s", field, filter);
+          if (transforms.containsKey(transformKey)) {
```

Comment:
existing issue, but it's more readable to do:

```
if (!transforms.containsKey(transformKey)) {
  transforms.put(transformKey, config.getTransform());
  continue;
}
...
```

This removes one layer of nesting and reduces the size of the code in the if block.